src/commands/infra/generate.ts

```diff
@@ -244,22 +243,51 @@ export const validateRemoteSource = async (
   logger.info(
     `Checking if source repo: ${safeLoggingUrl} has been already cloned to: ${sourcePath}.`
   );
-
   try {
-    // Check if .git folder exists in ${sourcePath}, if not, then clone
-    // if already cloned, 'git pull'
-    if (fs.existsSync(path.join(sourcePath, ".git"))) {
-      await gitFetchPull(sourcePath, safeLoggingUrl);
-    } else {
-      await gitClone(source, sourcePath);
-    }
+    // Clone remote source
+    await infraCommon.repoClone(sourcePath, source);
     // Checkout tagged version
     logger.info(`Checking out template version: ${version}`);
     await gitCheckout(sourcePath, version);
   } catch (err) {
-    logger.error(err);
-    // TOFIX: this error should be rethrown
-    // and if we do, infra-validations.sh will fail
+    if (err instanceof Error) {
```

Comment:
I suggest that we have small function for all the error handling code. Maybe break them down to 2 small functions (one for each case). So that they can be tested. Otherwise, code coverage is going to drop